src/views/vmnetworks/form/VMNetworkNewForm.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom-v5-compat';
+
+import { k8sCreate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  FormGroup,
+  Wizard,
+  WizardHeader,
+  WizardStep,
+} from '@patternfly/react-core';
+import ErrorAlert from '@utils/components/ErrorAlert';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { ClusterUserDefinedNetworkModel } from '@utils/models';
+import { isEmpty } from '@utils/utils';
+
+import NetworkDefinition from './components/NetworkDefinition';
+import ProjectMapping from './components/ProjectMapping';
+import { defaultFormValue, VMNetworkForm } from './constants';
+
+const VMNetworkNewForm: FC = () => {
+  const { t } = useNetworkingTranslation();
+  const [apiError, setError] = useState<Error>(null);
+
+  const methods = useForm<VMNetworkForm>({
+    defaultValues: defaultFormValue,
+  });
+
+  const {
+    formState: { isSubmitSuccessful, isSubmitting },
+    handleSubmit,
+    watch,
+  } = methods;
+
+  const name = watch('network.metadata.name');
+  const bridgeMapping = watch('network.spec.network.localnet.physicalNetworkName');
+  const namespaceSelector = watch('network.spec.namespaceSelector');
+  const projectList = watch('projectList');
+  const matchLabelCheck = watch('matchLabelCheck');
+
+  const emptyProjectList = projectList
+    ? isEmpty(namespaceSelector.matchExpressions)
+    : !matchLabelCheck && isEmpty(namespaceSelector.matchLabels);
+
+  const navigate = useNavigate();
```

Comment:
Could you please move this up to the top of the component since it doesn't rely on any of the other stuff that's currently above it?